src/Catel.Core/Catel.Core.Shared/Threading/Extensions/ReaderWriterLockSlimExtensions.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
```

Comment:
I see some repetition in this file. But I have no idea to reduce it without loosing readability.
I have also thought about overloads with parameters of those method but I'm not sure if they don't cause some bad code.